healthcheck/audit.go

```diff
@@ -0,0 +1,59 @@
+package healthcheck
+
+import (
+	"context"
+	"time"
+
+	"github.com/cloudtrust/common-service/v2/events"
+	log "github.com/cloudtrust/common-service/v2/log"
+)
+
+type auditEventsReporterChecker struct {
+	alias    string
+	reporter events.AuditEventsReporterModule
+	timeout  time.Duration
+	response HealthStatus
+	logger   log.Logger
+}
+
+func newAuditEventsReporterChecker(alias string, reporter events.AuditEventsReporterModule, timeout time.Duration, cacheDuration time.Duration, logger log.Logger) BasicChecker {
+	healthStatusType := "auditEventreporter"
+	response := HealthStatus{Name: &alias, Type: &healthStatusType, CacheDuration: cacheDuration}
+	response.connection("init")
+	response.stateUp()
+	return &auditEventsReporterChecker{
+		alias:    alias,
+		reporter: reporter,
+		timeout:  timeout,
+		response: response,
+		logger:   logger,
+	}
+}
+
+func (a *auditEventsReporterChecker) CheckStatus() HealthStatus {
+	if a.response.hasExpired() {
+		go a.updateStatus()
```

Comment:
You don't prevent refreshing status twice in the same time... but it's not blocking!